errors/apiErrors.ts

```diff
@@ -29,45 +56,53 @@ export function isSpecifiedError(
     code?: string;
   }
 ): boolean {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const error = (err && (err.cause as AxiosError<any>)) || err;
-  const statusCodeErr = !statusCode || error.response?.status === statusCode;
-  const categoryErr = !category || error.response?.data?.category === category;
-  const subCategoryErr =
-    !subCategory || error.response?.data?.subCategory === subCategory;
-  const errorTypeErr =
-    !errorType || error.response?.data?.errorType === errorType;
-  const codeError = !code || error.code === code;
+  if (!isHubSpotHttpError(err) && !isAxiosError(err.cause)) {
+    return false;
+  }
+
+  const {
+    data,
+    status,
+    code: actualCode,
+  } = extractHttpErrorFields(isHubSpotHttpError(err) ? err : err.cause);
+
+  const causedByAxiosError = isAxiosError(err.cause);
+
+  const statusCodeMatchesError = !statusCode || status === statusCode;
+  const categoryMatchesError = !category || data?.category === category;
+  const subCategoryMatchesError =
+    !subCategory || data?.subCategory === subCategory;
+  const errorTypeMatchesError = !errorType || data?.errorType === errorType;
+  const codeMatchesError = !code || actualCode === code;
 
   return (
-    error.isAxiosError &&
-    statusCodeErr &&
-    categoryErr &&
-    subCategoryErr &&
-    errorTypeErr &&
-    codeError
+    causedByAxiosError &&
+    statusCodeMatchesError &&
+    categoryMatchesError &&
+    subCategoryMatchesError &&
+    errorTypeMatchesError &&
+    codeMatchesError
   );
 }
 
-export function isMissingScopeError(err: Error | AxiosError): boolean {
+export function isMissingScopeError(err: Error | HubSpotHttpError): boolean {
   return isSpecifiedError(err, { statusCode: 403, category: 'MISSING_SCOPES' });
 }
 
-export function isGatingError(err: Error | AxiosError): boolean {
+export function isGatingError(err: Error | HubSpotHttpError): boolean {
   return isSpecifiedError(err, { statusCode: 403, category: 'GATED' });
 }
 
-export function isTimeoutError(err: Error | AxiosError): boolean {
+export function isTimeoutError(err: Error | HubSpotHttpError): boolean {
   return isSpecifiedError(err, { code: 'ETIMEDOUT' });
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function isApiUploadValidationError(err: AxiosError<any>): boolean {
+export function isApiUploadValidationError(err: HubSpotHttpError): boolean {
   return (
```

Comment:
Could this util make use of `isSpecifiedError`?

Reply:
Yep, it can use it for the first part of the conditional check.  Updated